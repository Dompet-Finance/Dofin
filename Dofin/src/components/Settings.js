'use strict'

import React from 'react'

import {
  View,
  Text,
  Button,
} from 'react-native'

class AppBar extends React.Component {
  render() {
    return (
      <View>
        <Text>
          AppBar
        </Text>
      </View>
    )
  }
}

class SettingItem extends React.Component {
  render() {
    return (
      <View>
        <Text>
          Setting A
        </Text>
        <Text>
          On / Off
        </Text>
      </View>
    )
  }
}

export class Settings extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <View>
        <AppBar />
        <SettingItem />
        <SettingItem />
        <SettingItem />
      </View>
    )
  }
}

const styles = {
}

export default Settings