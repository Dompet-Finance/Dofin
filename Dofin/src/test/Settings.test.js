'use strict'

import 'react-native'
import 'react-dom'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Settings from '../components/Settings.js'

describe('Test Settings component', () => {

  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Settings />)
  })

  test('renders correctly', () => {
    const tree = renderer.create(
      <Settings />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })

  test('should render appbar', () => {
    const child = wrapper.find('AppBar')
    expect(child).toHaveLength(1)
  })

  test('should render each setting item', () => {
    const child = wrapper.find('SettingItem').first()
    expect(child).toHaveLength(1)
  })

  test('should change state on change setting', () => {
    // manage state
  })

})

