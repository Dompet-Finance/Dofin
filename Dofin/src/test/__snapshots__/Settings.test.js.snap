// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Settings component renders correctly 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      AppBar
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Setting A
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      On / Off
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Setting A
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      On / Off
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Setting A
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      On / Off
    </Text>
  </View>
</View>
`;
